src/app/features/notification/notification-rule/notification-rule.component.ts

```diff
@@ -89,9 +96,13 @@ export class NotificationRuleComponent implements OnChanges {
   private parseChannelsToOptionsArray(channels?: {
     [key: string]: boolean;
   }): string[] {
-    return Object.entries(channels ?? [])
+    const parsedChannels = Object.entries(channels ?? [])
       .filter(([key, value]) => value === true)
       .map(([key, value]) => key);
+
+    return parsedChannels
+      .map((index) => this.notificationMethods[index]?.key)
+      .filter((key) => key !== undefined);
```

Comment:
which case makes this mapping necessary? Don't we already assign the notificationMethods' key as the value in each mat-option?

Reply:
Yes, we can assign the notificationMethods' key as the value in each mat-option. I have updated the code.

Reply:
but why do will still need that line `parsedChannels.map((index) => this.notificationMethods[index]?.key);` at all? Isn't that the same as the creating of parsedChannels above?

In any way, we should actually have some simple unit tests for this. Could you write one to check that a component.value is correctly parsed into the formControls after calling component.ngOnChanges({ value: true })  and another one that checks that the valueChange is emited (and has the correct parsed format) after one of the `form`'s formControls is being used with setValue()?

Reply:
The line `parsedChannels.map((index) => this.notificationMethods[index]?.key)` is needed because `parsedChannels` contains the indices where the values are true, but in `notificationMethods`, the key are in strings like as: `push`. So, first, in `parsedChannels`, we have the indices where the values are true, and then using this line, I extracted the `keys` from `notificationMethods` corresponding to the true indices and displayed them in the field.

Yes, sure. I will update the test cases for this.

Reply:
> The line `parsedChannels.map((index) => this.notificationMethods[index]?.key)` is needed because `parsedChannels` contains the indices where the values are true, but in `notificationMethods`, the key are in strings like as: `push`. So, first, in `parsedChannels`, we have the indices where the values are true, and then using this line, I extracted the `keys` from `notificationMethods` corresponding to the true indices and displayed them in the field.

no, this is not how the `MatSelect` component works and outputs values.
You can put a breakpoint or console.log into our updateValue() method to see how the form's value actually looks (it's an array of the keys of the active mat-options, i.e. `["push"]` or `[]` for us)

Reply:
I rechecked the notification method `mat-select`. The issue in this line `for (let option in options ?? [])` Now we’re using `of` instead of `in`. Now, it's working fine.
I have update the code. Please take a look once.